ta_storage/tests/test_bq.py

```diff
@@ -1,485 +1,381 @@
+            preserving_proto_field_name=True,
         )
+        for testrun_bytes in mock_bigquery_service.mock_calls[0][1][3]
+    ]
+    assert snapshot("json") == sorted(testruns_written, key=lambda x: x["name"])
 
 
-@pytest.mark.skip(reason="need creds")
-def test_bq_pr_comment():
-    bq = BQDriver()
+@pytest.mark.django_db(transaction=True, databases=["test_analytics"])
+def test_write_testruns_with_flake(mock_bigquery_service, mock_config, snapshot):
+    driver = BQDriver(repo_id=1)
+    timestamp = int(
+        datetime.fromisoformat("2025-01-01T00:00:00Z").timestamp() * 1000000
+    )
 
-    if (
-        bq.bq_service.query(
-            "select * from `test_dataset.testruns` where repoid = 2 limit 1"
-        )
-        == []
-    ):
-        populate_pr_comment_testruns(bq)
+    flake = Flake.objects.create(
+        repoid=1,
+        test_id=calc_test_id("test_suite", "TestClass", "test_something"),
+        flags_id=calc_flags_hash(["unit"]),
+        fail_count=1,
+        count=1,
+        recent_passes_count=1,
+        start_date=datetime.now(timezone.utc),
+    )
+    flake.save()
 
-    pr_agg = bq.pr_comment_agg(repoid=2, commit_sha="abcde")
-    assert pr_agg == [
+    testruns: list[test_results_parser.Testrun] = [
         {
-            "commit_sha": "abcde",
-            "ct_passed": 6,
-            "ct_failed": 3,
-            "ct_skipped": 0,
-            "ct_flaky_failed": 0,
+            "name": "test_something",
+            "classname": "TestClass",
+            "testsuite": "test_suite",
+            "computed_name": "TestClass.test_something",
+            "outcome": "failure",
+            "failure_message": "assertion failed",
+            "duration": 1.5,
+            "filename": "test_file.py",
+            "build_url": None,
         }
     ]
 
-    pr_fail = bq.pr_comment_fail(repoid=2, commit_sha="abcde")
-    assert len(pr_fail) == 3
-    assert {t["computed_name"] for t in pr_fail} == {
-        "pr_computed_name_1",
-    }
-    assert {t["failure_message"] for t in pr_fail} == {"hi"}
-    assert {tuple(t["flags"]) for t in pr_fail} == {
-        ("flag_1",),
-        ("flag_2",),
-        ("flag_0",),
-    }
+    driver.write_testruns(
+        timestamp=timestamp,
+        commit_sha="abc123",
+        branch_name="main",
+        upload_id=1,
+        flag_names=["unit"],
+        framework="pytest",
+        testruns=testruns,
+    )
 
+    mock_bigquery_service.write.assert_called_once_with(
+        "test-dataset", "test-table", ta_testrun_pb2, ANY
+    )
 
-def populate_testruns_for_upload_testruns(dbsession, bq: BQDriver):
-    testruns = []
+    testruns_written = [
+        MessageToDict(
+            ta_testrun_pb2.TestRun.FromString(testrun_bytes),
+            preserving_proto_field_name=True,
+        )
+        for testrun_bytes in mock_bigquery_service.mock_calls[0][1][3]
+    ]
+    assert snapshot("json") == sorted(testruns_written, key=lambda x: x["name"])
```

Comment:
you are asserting that the mocked function is being called with the arguments you put in above.
IMO the value of such tests is very low. All you are asserting here is that the serialization to protobuf is working correctly?